Tools/LyTestTools/tests/unit/test_o3de_multi_test_framework.py

```diff
@@ -0,0 +1,1184 @@
+
+        under_test = multi_test_framework.Result.Unknown(
+            mock_log_attribute, mock_test_spec, mock_output, mock_extra_info, mock_log_output)
+
+        assert mock_output in str(under_test)
+        assert mock_extra_info in str(under_test)
+
+
+class TestAbstractTestSuite(unittest.TestCase):
+
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.kill_all_ly_processes')
+    def test_TestData_ValidAP_TeardownAPOnce(self, mock_kill_processes):
+        mock_abstract_test_suite = multi_test_framework.AbstractTestSuite()
+        mock_test_data_generator = mock_abstract_test_suite._collected_test_data(mock.MagicMock())
+        mock_asset_processor = mock.MagicMock()
+        for test_data in mock_test_data_generator:
+            test_data.asset_processor = mock_asset_processor
+        mock_asset_processor.teardown.assert_called()
+        assert test_data.asset_processor is None
+        mock_kill_processes.assert_called_once_with(include_asset_processor=True)
+
+    @mock.patch('ly_test_tools.o3de.editor_test_utils.kill_all_ly_processes')
+    def test_TestData_NoAP_NoTeardownAP(self, mock_kill_processes):
+        mock_abstract_test_suite = multi_test_framework.AbstractTestSuite()
+        mock_test_data_generator = mock_abstract_test_suite._collected_test_data(mock.MagicMock())
+        for test_data in mock_test_data_generator:
+            test_data.asset_processor = None
+        mock_kill_processes.assert_called_once_with(include_asset_processor=False)
+
+    @mock.patch('ly_test_tools.o3de.multi_test_framework.AbstractTestSuite.filter_session_shared_tests')
+    def test_PytestCustomModifyItems_FunctionsMatch_AddsRunners(self, mock_filter_tests):
+        class MockTestSuite(multi_test_framework.AbstractTestSuite):
+            pass
+        mock_func_1 = mock.MagicMock()
+        mock_test = mock.MagicMock()
+        runner_1 = multi_test_framework.AbstractTestSuite.Runner('mock_runner_1', mock_func_1, [mock_test])
+        mock_run_pytest_func = mock.MagicMock()
+        runner_1.run_pytestfunc = mock_run_pytest_func
+        mock_result_pytestfuncs = [mock.MagicMock()]
+        runner_1.result_pytestfuncs = mock_result_pytestfuncs
+        mock_items = []
+        mock_items.extend(mock_result_pytestfuncs)
+
+        MockTestSuite._runners = [runner_1]
+        mock_test_1 = mock.MagicMock()
+        mock_test_2 = mock.MagicMock()
+        mock_filter_tests.return_value = [mock_test_1, mock_test_2]
+
+        MockTestSuite.pytest_custom_modify_items(mock.MagicMock(), mock_items, mock.MagicMock())
+        assert mock_items == [mock_run_pytest_func, mock_result_pytestfuncs[0]]
+
+    def test_GetSingleTests_NoSingleTests_EmptyList(self):
+        class MockTestSuite(multi_test_framework.AbstractTestSuite):
+            pass
+        mock_test_suite = MockTestSuite()
+        tests = mock_test_suite.get_single_tests()
+        assert len(tests) == 0
+
+    def test_GetSingleTests_OneSingleTests_ReturnsOne(self):
```

Comment:
minor: Tests could be added which populate the suite with concrete EditorXYZTest and MaterialEditorXYZTest subclasses, and assert similar functionality as this test. I expect this may be covered by making any necessary changes to the hierarchy of classes such as SingleTest / EditorSingleTest / MaterialEditorSingleTest.

This issue is effectively hidden by not having abstract enforcement from ABC on the "abstract" classes